components/AppBar/examples/AppBarWrappingUITableViewControllerExample.m

```diff
@@ -0,0 +1,101 @@
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "MaterialAppBar.h"
+#import "MaterialAppBar+ColorThemer.h"
+#import "MaterialAppBar+TypographyThemer.h"
+
+@interface AppBarWrappingUITableViewControllerExample : UIViewController <UITableViewDataSource>
+
+@property(nonatomic, strong) MDCAppBarContainerViewController *appBarContainerViewController;
+@property(nonatomic, strong) MDCSemanticColorScheme *colorScheme;
+@property(nonatomic, strong) MDCTypographyScheme *typographyScheme;
+
+@end
+
+@implementation AppBarWrappingUITableViewControllerExample
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  UITableViewController *tableViewController =
+      [[UITableViewController alloc] initWithStyle:UITableViewStylePlain];
+  tableViewController.title = @"Wrapped table view";
+  self.appBarContainerViewController =
+      [[MDCAppBarContainerViewController alloc] initWithContentViewController:tableViewController];
+
+  tableViewController.tableView.dataSource = self;
+  [tableViewController.tableView registerClass:[UITableViewCell class]
+                        forCellReuseIdentifier:@"cell"];
+
+  self.appBarContainerViewController.appBar.headerViewController.headerView.trackingScrollView =
+      tableViewController.tableView;
+
+  [MDCAppBarColorThemer applySemanticColorScheme:self.colorScheme
+                                        toAppBar:self.appBarContainerViewController.appBar];
+  [MDCAppBarTypographyThemer applyTypographyScheme:self.typographyScheme
+                                          toAppBar:self.appBarContainerViewController.appBar];
+
+  // Need to update the status bar style after applying the theme.
+  [self setNeedsStatusBarAppearanceUpdate];
+
+  [self addChildViewController:self.appBarContainerViewController];
+  self.appBarContainerViewController.view.frame = self.view.bounds;
+  [self.view addSubview:self.appBarContainerViewController.view];
+  [self.appBarContainerViewController didMoveToParentViewController:self];
+}
+
+- (UIViewController *)childViewControllerForStatusBarStyle {
+  return self.appBarContainerViewController;
+}
+
+#pragma mark - UITableViewDataSource
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  return 100;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView
+         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+  UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"
+                                                          forIndexPath:indexPath];
+  cell.textLabel.text = [@(indexPath.row) description];
+  return cell;
+}
+
+@end
+
+@implementation AppBarWrappingUITableViewControllerExample (CatalogByConvention)
+
++ (NSArray *)catalogBreadcrumbs {
+  return @[ @"App Bar", @"Wrapped table view" ];
+}
+
++ (BOOL)catalogIsPrimaryDemo {
+  return NO;
+}
+
+- (BOOL)catalogShouldHideNavigation {
+  return YES;
+}
+
++ (BOOL)catalogIsPresentable {
+  return YES;
```

Comment:
Are you sure you want this in the catalog app? An 100 rows tableview seems dragon-y.

Reply:
It's what we'd consider a "typical use" of the App Bar (a table view wrapped by an App Bar) so we do want it shown in the main catalog (everything in the MDCCatalog is meant to demonstrate primary/secondary use cases that allow us to evaluate accessibility).

Reply:
ah ok. i guess i have a slight misunderstanding on the catalog's purpose then, we can sync up offline.